flake.nix

```diff
@@ -97,6 +99,13 @@
             enableDebug = true;
             runTests = false;
           });
+          proof-producer-static-build = (pkgs.callPackage ./proof-producer.nix {
+            runTests = false;
+            enableDebug = false;
+            staticBuild = true;
+            boost = pkgs.pkgsStatic.boost.override{ enableShared = false;};
+            protobuf = (pkgs.pkgsStatic.protobuf.override { enableShared = false;});
```

Comment:
Maybe easier would be to override boost and protobuf conditionally depending on staticBuild flag inside the derivation?
You can also set PROOF_PRODUCER_STATIC_BINARIES, BUILD_SHARED_LIBS and Boost_USE_STATIC_LIBS flags conditionally

Reply:
May be just use static build allays for all derivations?
I guess we have no special cases where required dynamic linkage.

Reply:
I thought about it, but now `proof-producer.nix` doesn't have access to all packages, and in this case it must have it to choose between `pkgs` and `pkgs.static`. So I will have to make something like `pkgs.callPackage ./proof-producer.nix { pkgs = pkgs; ... }` which seems weird to me.

Reply:
Konstantin mentioned ASan build where it's helpful to have shared libraries (can't say why, I am not an expert in sanitizers).

Reply:
yes, according our dissolution, let's keep both build options: static is main (product), dynamic for additional checks